core/config/docs/docs_test.go

```diff
@@ -124,3 +124,21 @@ func assertTOML[T any](t *testing.T, fallback, docs T) {
 	fs, ds := string(fb), string(db)
 	assert.Equal(t, fs, ds, diff.Diff(fs, ds))
 }
+
+func assertFieldsNotNil(t *testing.T, c chainlink.Config) {
```

Comment:
Hacks should be explicit, visible, and documented:

```suggestion
// Exclude TRON-specific custom config from the nil-check (required only for TRON)
// TODO: restructure with TRON-specific configs
func assertFieldsNotNil_ExceptHTTPURLExtraWrite(t *testing.T, c chainlink.Config) {
```